tutorials/mongodb-atlas-visionapi-nodejs/index.md

```diff
@@ -0,0 +1,680 @@
+
+    ![image](images/image16.png)
+
+7.  You should see a CloudShell window similar to this open...
+
+    ![image](images/image7.png)
+
+8.  Install Node JS and Express by executing following commands:
+
+        sudo apt-get update
+        sudo apt-get install -y nodejs
+        sudo apt-get install npm
+        curl -sL https://deb.nodesource.com/setup_8.x | sudo -E bash -
+        sudo apt-get install -y build-essential
+        npm install nconf
+        npm install express --save
+        npm install mongodb@2.2.33 --save
+
+9.  Install vision library:
+
+        npm install --save @google-cloud/vision
+   
+10. Install storage library:
+
+        npm install --save @google-cloud/storage
+
+11. Enable cloud vision API:
+
+        gcloud services enable vision.googleapis.com
+
+12. Create bucket with a name for your project.  Replace [bucketname] with your unique name.  Note buckets have to have a globally unique name to be accepted:
+
+        gsutil mb gs://[bucketname]
+
+        ![image](images/image26.png)        
+
+13. Make bucket public:
+
+        gsutil defacl set public-read gs://[bucketname]
+
+        ![image](images/image29.png)   
+
+14. Let's switch back to the console and navigate to our new Google Cloud Storage bucket, by entering “Storage” in the search query and selecting the recommendation.  
+    Please verify your new bucket exists and select it
+    
+    ![image](images/image36.png)
+
+15. Once selected, you can drag and drop a few sample images for our demo server to process:   
+
+    ![image](images/image8.png)
+
+    ![image](images/image10.png)    
+
+16. Switch back to your SSH window and lets create a folder named: “nodejs-express-mongodb”:
+
+        mkdir nodejs-express-mongodb
+
+17. Change directory to this folder:
+
+        cd nodejs-express-mongodb
+
+18. Preparing the app
+
+        1.  Initialize a package.json file with the following command:
+
+                npm init
+
+                ![image](images/image21.png)
+
+            Setup a new node project. You can press enter to accept the default settings, but when you get to entry point, type in “server.js” 
+
+                ![image](images/image17.png)
+
+            Enter “yes” when prompted and press enter
+
+        2.  We need to create a public folder for our server and select it, by entering the following commands:
+
+                mkdir public
+                cd public
+
+                ![image](images/image2.png)
+
+        3.  Let’s create the client side file using the following command and copying the following script content into the editor. Save the file using Ctrl + X:
+
+                nano client.js
+
+                        ****************************************
+
+                        console.log('Client-side code running');
+
+                        const button = document.getElementById('myButton');
+
+                        button.addEventListener('click', function(e) {
```

Comment:
Remove all the extra newlines, this code is hard to read, apply throughout

Reply:
I cleaned the code up by removing extraneous new lines and making indentation more consistent.